lib/ansible/executor/task_executor.py

```diff
@@ -817,10 +817,12 @@ def _start_connection(self):
         (stdout, stderr) = p.communicate()
         stdin.close()
 
-        if p.returncode == 0:
-            result = json.loads(stdout)
-        else:
-            result = json.loads(stderr)
+        out = stdout if p.returncode == 0 else stderr
+        if br"##RESPONSE##:" not in out:
+            return None
+
+        out = out.split(br"##RESPONSE##:")[1]
+        result = json.loads(to_text(out.rstrip(), errors='surrogate_then_replace'))
```

Comment:
please use out.strip() to handle leading whitespaces from messages as well

Reply:
Fixed